altair/vegalite/v5/api.py

```diff
@@ -3357,14 +3380,14 @@ def __init__(self, data=Undefined, hconcat=(), **kwargs):
         self.data, self.hconcat = _combine_subchart_data(self.data, self.hconcat)
         self.params, self.hconcat = _combine_subchart_params(self.params, self.hconcat)
 
-    def __ior__(self, other: NonNormalizedSpec) -> Self:
```

Comment:
Why do we have to remove this? Do we ever pass anything other than a `NonNormalizedSpec`?

Reply:
> Why do we have to remove this? Do we ever pass anything other than a `NonNormalizedSpec`?

I was really confused about this annotation.

All `ChartType` classes would fail this, since they derive from `TopLevelSpec`

https://github.com/vega/altair/blob/9afd374a83f954e79a5fd6e4f8a8a9c4e5381c04/altair/vegalite/v5/schema/core.py#L23616

Reply:
@binste to demonstrate the effects of using `NonNormalizedSpec` here, vs no annotation in `TopLevelMixin` 

![image](https://github.com/user-attachments/assets/ca1ac3f3-c1f4-471e-a1e1-8534e220f747)

Whereas simply removing this annotation

![image](https://github.com/user-attachments/assets/ebda4260-919d-4ef7-8fcf-6e3708e69641)

---

So the current annotation results in `pylance` users getting a return type of `Any` and requires them to add a type ignore

Reply:
Hmm I can't replicate why I've added this in the first place. Thanks for the explanation and fixing it! Let's remove it as you've already done.